### PR TITLE
Add Microsoft.Databricks 2024-09-01-preview version

### PR DESCRIPTION
**Pull request by curiekim00**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
     "title": "DatabricksAccessConnectorClient",
-    "version": "2024-05-01",
+    "version": "2024-09-01-preview",
```
Comment (scottmacchia-msft) [inline specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json:5]: @curiekim00 Can you check with Subbu or Yuqing as to whether we need to update Access Connector?

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/databricks.json [hunk 2]
```diff
@@ -1153,7 +1153,8 @@
       "enum": [
         "NONE",
         "HIPAA",
-        "PCI_DSS"
+        "PCI_DSS",
+        "CYBER_ESSENTIAL_PLUS"
```
Comment (scottmacchia-msft) [inline specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/databricks.json:1157]: @curiekim00 Is there a precedence for including location restrictions in Swagger specs? We should at least check if this exists.

Comment (curiekim00) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/databricks.json:1157]: Hi @scottmacchia-msft I have checked that there was no precedence for including location restrictions and have double checked with the Breaking Change reviewers that it is not possible to add validations in the spec, and suggested just documenting this restriction

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json (hunk 1 above)
Comment (curiekim00) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json]: Hi @scottmacchia-msft checked with Yuqing, there is no pending changes in the Access Connector, so I have reverted the Access Connector version update and pointed to the 2024-05-01 version, will double check with Subbu again when he is back

**Comment by mentat9** [comment]
@curiekim00 - Please use the template above to create/update your PR with the property intake.

**Comment by mentat9** [comment]
@curiekim00 - Please respond to reviewer comments (https://github.com/Azure/azure-rest-api-specs/pull/30175#issuecomment-2297682301) before removing the ARMChangesRequested label.

**Comment by curiekim00** [comment]
Hi @mentat9 Misunderstood and thought you were referring to the "next steps to merge" last time, I have updated the template above, could you please take a look at the PR?

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json (hunk 1 above)
Comment (mentat9) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json]: You should move all your APIs forward together per uniform versioning. This is not an absolute requirement yet, but some SDKs can't be generated with mixed API versions, so uniform versioning is highly recommended.

Moving everything forward with each new API version makes consuming your APIs and SDKs much easier for customers.

**Comment by curiekim00** [comment]
Hi @mentat9 I saw that you have added the "ArmChangesRequested" tag again without any comment, I have the PR updated according to the template, could you please explain what is missing?

**Comment by mentat9** [comment]
@curiekim00 - Unresolved comment thread to make my comments visible.

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json (hunk 1 above)
Comment (curiekim00) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json:5]: Hi @mentat9 I have upgraded the API version for Access Connector as well, but it looks like now the bot has added the Breaking Change tag despite it not being breaking change before the removal.

Comment (mentat9) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-09-01-preview/accessconnector.json:5]: Yes, the Python SDK team will help you with that. The fix guide link in the Next Steps to Merge comment tells you how to proceed.

**Review by scottmacchia-msft — APPROVED** [review]

**Pull request merged**